microsoft/testsuites/vm_extensions/vmsnapshot_extension.py

```diff
@@ -89,8 +90,25 @@ def verify_vmsnapshot_extension(
                     parameters={},
                 )
                 wait_operation(response, time_out=600)
-                log.info(f"restore point {restore_point} created")
-                break
+                # check the status of rp and validate the result.
+                return_value = self.get_restore_point(
+                    log,
+                    environment,
+                    resource_group_name,
+                    restore_point_collection,
+                    restore_point,
+                )
+                if return_value == 0:
```

Comment:
The return value should be more meaningful, instead of a number. In this case, it looks like the method can raise different exceptions directly, if it needs to return 1 or 2. So it doesn't need to return values, if everything works find.